Client/Frontend/Browser/TopTabsViewController.swift

```diff
@@ -0,0 +1,714 @@
+            make.trailing.equalTo(view)
+            make.size.equalTo(UIConstants.ToolbarHeight)
+        }
+        tabsButton.snp_makeConstraints { make in
+            make.centerY.equalTo(view)
+            make.trailing.equalTo(newTab.snp_leading)
+            make.size.equalTo(UIConstants.ToolbarHeight)
+        }
+        privateTab.snp_makeConstraints { make in
+            make.centerY.equalTo(view)
+            make.leading.equalTo(view)
+            make.size.equalTo(UIConstants.ToolbarHeight)
+        }
+        topTabFader.snp_makeConstraints { make in
+            make.top.bottom.equalTo(view)
+            make.leading.equalTo(privateTab.snp_trailing).offset(-TopTabsUX.FaderPading)
+            make.trailing.equalTo(tabsButton.snp_leading).offset(TopTabsUX.FaderPading)
+        }
+        collectionView.snp_makeConstraints { make in
+            make.top.bottom.equalTo(view)
+            make.leading.equalTo(privateTab.snp_trailing).offset(-TopTabsUX.CollectionViewPadding)
+            make.trailing.equalTo(tabsButton.snp_leading).offset(TopTabsUX.CollectionViewPadding)
+        }
+        
+        view.backgroundColor = UIColor.blackColor()
+    }
+    
+    func updateTabCount(count: Int, animated: Bool = true) {
+        self.tabsButton.updateTabCount(count, animated: animated)
+    }
+    
+    func tabsClicked() {
+        delegate.topTabsPressTabs()
+    }
+    
+    override func viewDidAppear(animated: Bool) {
+        collectionView.reloadData()
+        self.scrollToCurrentTab(false)
+        super.viewDidAppear(animated)
+    }
+    
+    func newTabClicked() {
+        if let currentTab = tabManager.selectedTab, let index = tabsToDisplay.indexOf(currentTab),
+            let cell  = collectionView.cellForItemAtIndexPath(NSIndexPath(forItem: index, inSection: 0)) as? TopTabCell {
+            cell.selectedTab = false
+            if index > 0 {
+                cell.seperatorLine = true
+            }
+        }
+        delegate.topTabsPressNewTab()
+        collectionView.performBatchUpdates({ _ in
+            let count = self.collectionView.numberOfItemsInSection(0)
+            self.collectionView.insertItemsAtIndexPaths([NSIndexPath(forItem: count, inSection: 0)])
+            }, completion: { finished in
+                if finished {
+                    self.scrollToCurrentTab()
+                }
+        })
+    }
+    
+    func privateTabClicked() {
+        delegate.topTabsPressPrivateTab()
+        self.collectionView.reloadData()
+        scrollToCurrentTab(false)
+    }
+    
+    func closeTab() {
+        delegate.topTabsPressTabs()
+    }
+    
+    func reloadFavicons() {
+        self.collectionView.reloadData()
+    }
+    
+    func scrollToCurrentTab(animated: Bool = true) {
+        guard let currentTab = tabManager.selectedTab, let index = tabsToDisplay.indexOf(currentTab) else {
+            return
+        }
+        guard !collectionView.frame.isEmpty else {
```

Comment:
Nit: You can merge this boolean check with the above guard by adding a `where` clause:

```swift
guard let currentTab = tabManager.selectedTab, let index = tabsToDisplay.indexOf(currentTab) where !collectionView.frame.isEmpty else {
    return
}
```